tests/tests/tier0/hirte-agent-started-and-connected/test_agent_started_and_connected.py

```diff
@@ -1,56 +1,36 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
-import platform
 import pytest
-import socket
+from typing import Dict
 
-from podman.domain.containers import Container
+from hirte_test.test import HirteTest
+from hirte_test.container import HirteContainer
+from hirte_test.config import HirteControllerConfig, HirteNodeConfig
 
-from hirte_tests.gather import journal
-from hirte_tests.provision import containers
 
+@pytest.mark.timeout(5)
+def test_agent_foo_startup(
+        hirte_test: HirteTest,
+        hirte_ctrl_default_config: HirteControllerConfig,
+        hirte_node_default_config: HirteNodeConfig):
 
-@pytest.fixture(autouse=True)
-def handle_env(
-    hirte_controller_ctr: Container,
-    hirte_node_foo_ctr: Container,
-    tmt_test_data_dir: Container,
-):
-    # initialize configuration
-    containers.put_files(hirte_controller_ctr,
-                         'controller.conf.d', '/etc/hirte/hirte.conf.d')
-    containers.put_files(hirte_node_foo_ctr,
-                         'node-foo.conf.d', '/etc/hirte/agent.conf.d')
+    hirte_node_default_config.node_name = "node-foo"
+    hirte_ctrl_default_config.allowed_node_names = [hirte_node_default_config.node_name]
 
-    # TODO: find out a better way how to set ManagerHost
-    hirte_node_foo_ctr.exec_run(
-        'sed -i "s/@MANAGER_HOST@/{manager_host}/" /etc/hirte/agent.conf.d/03-manager.conf'.format(
-            manager_host=socket.gethostbyname(platform.node()),
-        )
-    )
+    hirte_test.set_hirte_controller_config(hirte_ctrl_default_config)
+    hirte_test.add_hirte_node_config(hirte_node_default_config)
 
-    # start services
-    hirte_controller_ctr.exec_run('systemctl start hirte')
-    hirte_node_foo_ctr.exec_run('systemctl start hirte-agent')
+    def foo_startup_verify(ctrl: HirteContainer, nodes: Dict[str, HirteContainer]):
```

Comment:
Can this function be a top level function? It seems to me that it would be more readable by being top level rather than being inside `test_agent_foo_startup` function ...

Reply:
The idea is to really isolate these functions "from the outside". I'd prefer using lambdas, but those are not really good to use in python, I think.

Reply:
I know, but you should import a specific python test  file into a different test file. It just seems to me that those verify functions on top level are more readable for test developers.

And you can add `_` prefix to the verify function name, which is a python standard for making function private ...

Reply:
Yes, it should not, but it still can - and the underscore is also just a "you should not", but it doesn't prevent it as far as I know. Nesting the function is a nice way to achieve this and increases the cohesion. 
But if its more readable for test developers, we can move it one level up.

Reply:
> Yes, it should not, but it still can - and the underscore is also just a "you should not", but it doesn't prevent it as far as I know. Nesting the function is a nice way to achieve this and increases the cohesion. But if its more readable for test developers, we can move it one level up. 

Yes, I know `_` is only an agreement, but not everything can be forced :-)

> Edit: Moved it

Thanks!